### ISSUE TITLE
実機でも動作する誤差逆伝播の実装

### ISSUE THREAD

**Issue by ikanago** [Qulacs-Osaka/scikit-quri#1]
## 候補
- Quantum Circuit Parameters Learning with Gradient Descent Using Backpropagation [[0]](https://arxiv.org/pdf/1910.14266)
- Training deep quantum neural networks [[1]](https://www.nature.com/articles/s41467-020-14454-2.pdf)
- Deep quantum neural networks equipped with backpropagation on a superconducting processor [[2]](https://arxiv.org/pdf/2212.02521)

## [0]
- 以下の式で微分を求められるらしい
  - $c_\theta^j$ は $\ket{j}$ の確率振幅で、 $p_\theta^j = |c_\theta^j|^2$
  - 確率振幅が必要なので実機では動かないはず...
$$\frac{\partial L}{\partial \theta} = 2\mathrm{Re} \left[ \frac{\partial L}{\partial p_\theta^j} \frac{\partial p_\theta^j}{\partial c_\theta^j} \frac{\partial c_\theta^j}{\partial \theta} \right]$$

## [1]
- 順伝播の一部の計算を以下のように行う
  - 部分トレースの計算が必要
    - 実機でできる？
  - パラメトリックゲートを使うわけではない
- 著者による解説：https://www.youtube.com/watch?v=_M2GQAknykg
$$\rho^{\mathrm{out}} = \mathrm{tr}\_{\mathrm{in,hid}} \left( U\rho_{\mathrm{in}}\otimes\ket{0\cdots 0}_{\mathrm{hid,out}}\bra{0\cdots 0}U^\dagger \right)$$

## [2]
- [1] とベースは同じ

**Comment by ikanago** [comment]
サーベイ論文的なやつ：Quantum computing models for artificial neural networks [[3]](https://iopscience.iop.org/article/10.1209/0295-5075/134/10002/pdf)

**Comment by ikanago** [comment]
- A Universal Training Algorithm for Quantum Deep Learning [[4]](https://arxiv.org/pdf/1806.09729)

## [4]
- 論文が長い
- パラメータを位相で表現する？
